modules/table/src/test/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandlerTest.java

```diff
@@ -0,0 +1,195 @@
+
+        sortedInnerStorage = new TestSortedIndexStorage(PARTITION_ID, new StorageSortedIndexDescriptor(sortedIndexId, List.of(
+                new StorageSortedIndexColumnDescriptor("INTVAL", NativeTypes.INT32, false, true),
+                new StorageSortedIndexColumnDescriptor("STRVAL", NativeTypes.STRING, false, true)
+        )));
+
+        TableSchemaAwareIndexStorage sortedIndexStorage = new TableSchemaAwareIndexStorage(
+                sortedIndexId,
+                sortedInnerStorage,
+                USER_INDEX_BINARY_TUPLE_CONVERTER
+        );
+
+        hashInnerStorage = new TestHashIndexStorage(PARTITION_ID, new StorageHashIndexDescriptor(hashIndexId, List.of(
+                new StorageHashIndexColumnDescriptor("INTVAL", NativeTypes.INT32, false),
+                new StorageHashIndexColumnDescriptor("STRVAL", NativeTypes.STRING, false)
+        )));
+
+        TableSchemaAwareIndexStorage hashIndexStorage = new TableSchemaAwareIndexStorage(
+                hashIndexId,
+                hashInnerStorage,
+                USER_INDEX_BINARY_TUPLE_CONVERTER
+        );
+
+        lock = spy(new LockByRowId());
+        storage = spy(new TestMvPartitionStorage(PARTITION_ID, lock));
+
+        Map<Integer, TableSchemaAwareIndexStorage> indexes = Map.of(
+                pkIndexId, pkStorage,
+                sortedIndexId, sortedIndexStorage,
+                hashIndexId, hashIndexStorage
+        );
+
+        TestPartitionDataStorage partitionDataStorage = new TestPartitionDataStorage(tableId, PARTITION_ID, storage);
+
+        indexUpdateHandler = spy(new IndexUpdateHandler(DummyInternalTableImpl.createTableIndexStoragesSupplier(indexes)));
+
+        storageUpdateHandler = new StorageUpdateHandler(
+                PARTITION_ID,
+                partitionDataStorage,
+                indexUpdateHandler,
+                storageUpdateConfiguration
+        );
+    }
+
+    @Test
+    void testUpdateAllBatchedTryLockFailedOnce() {
+        UUID txUuid = UUID.randomUUID();
+
+        HybridTimestamp commitTs = CLOCK.now();
+
+        BinaryRow row1 = binaryRow(new TestKey(1, "foo1"), new TestValue(2, "bar"));
+        BinaryRow row2 = binaryRow(new TestKey(3, "foo3"), new TestValue(4, "baz"));
+        BinaryRow row3 = binaryRow(new TestKey(5, "foo5"), new TestValue(7, "zzu"));
+
+        TablePartitionId partitionId = new TablePartitionId(333, PARTITION_ID);
+
+        TimedBinaryRow tb1 = new TimedBinaryRow(row1, null);
+        TimedBinaryRow tb2 = new TimedBinaryRow(row2, null);
+        TimedBinaryRow tb3 = new TimedBinaryRow(row3, null);
+
+        UUID id1 = UUID.randomUUID();
+        UUID id2 = UUID.randomUUID();
+        UUID id3 = UUID.randomUUID();
+
+        Map<UUID, TimedBinaryRow> rowsToUpdate = Map.of(
+                id1, tb1,
+                id2, tb2,
+                id3, tb3
+        );
+        // All batches consist of one row.
+        storageUpdateConfiguration.batchByteLength().update(0);
```

Comment:
This method returns future, but you're not waiting for the result. Please use something like `assertThat(fut, willCompleteSuccessfully())` on it

Reply:
Anyway, `batchLength = 0` should effectively lead to batches of 1 row, right? How could you even attempt `tryLock` if first time it's `lock`? Something's wrong with this test, or maybe with my understanding of it. Does `0` means `unlimited`? Minimal value, according to schema, is 1, is it documented? Maybe this test has a data race and that's it?
Setting `Integer.MAX_VALUE` wouldn't lead to such questions.